### ISSUE TITLE
finding dependency via CMake gives gcc version error

### ISSUE THREAD

**Issue by astinus87** [mesonbuild/meson#7545]
I have a library called Magnum that I want Meson to find via CMake and library has the FindMagnum.cmake file. When `meson build` is run Meson finds the FindMagnum.cmake but I get "corrade cannot be used with gcc < 4.8.1" error listed in meson-log.txt. I don't know why this happens because my gcc version is 10.1.0. Also meson correctly reports gcc version as 10.1.0 when `meson build` is run.

Meson Version: 0.54.3
Cmake Version: 3.18.1
Kernel: 5.7.12-arch1-1
Distro: Arch Linux

meson-log.txt:
```
Build started at 2020-08-05T13:19:51.409737
Main binary: /usr/bin/python
Build Options: 
Python system: Linux
The Meson build system
Version: 0.54.3
Source dir: /home/alp/practice/magnum
Build dir: /home/alp/practice/magnum/build
Build type: native build
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.
Project name: test
Project version: undefined
None of 'CXX' are defined in the environment, not changing global flags.
None of 'CXXFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CXX_LD' are defined in the environment, not changing global flags.
Sanity testing C++ compiler: c++
Is cross compiler: False.
Sanity check compiler command line: c++ /home/alp/practice/magnum/build/meson-private/sanitycheckcpp.cc -o /home/alp/practice/magnum/build/meson-private/sanitycheckcpp.exe -pipe -D_FILE_OFFSET_BITS=64
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/alp/practice/magnum/build/meson-private/sanitycheckcpp.exe
C++ compiler for the build machine: c++ (gcc 10.1.0 "c++ (GCC) 10.1.0")
C++ linker for the build machine: c++ ld.bfd 2.34.0
None of 'AR' are defined in the environment, not changing global flags.
None of 'CXX' are defined in the environment, not changing global flags.
None of 'CXXFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CXX_LD' are defined in the environment, not changing global flags.
Sanity testing C++ compiler: c++
Is cross compiler: False.
Sanity check compiler command line: c++ /home/alp/practice/magnum/build/meson-private/sanitycheckcpp.cc -o /home/alp/practice/magnum/build/meson-private/sanitycheckcpp.exe -pipe -D_FILE_OFFSET_BITS=64
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/alp/practice/magnum/build/meson-private/sanitycheckcpp.exe
C++ compiler for the host machine: c++ (gcc 10.1.0 "c++ (GCC) 10.1.0")
C++ linker for the host machine: c++ ld.bfd 2.34.0
None of 'AR' are defined in the environment, not changing global flags.
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Pkg-config binary for MachineChoice.HOST is not cached.
None of 'PKG_CONFIG' are defined in the environment, not changing global flags.
Pkg-config binary missing from cross or native file, or env var undefined.
Trying a default Pkg-config fallback at pkg-config
Trying pkg-config binary pkg-config for machine MachineChoice.HOST at ['/usr/bin/pkg-config']
Found pkg-config: /usr/bin/pkg-config (1.7.3)
Determining dependency 'Magnum' with pkg-config executable '/usr/bin/pkg-config'
PKG_CONFIG_PATH: 
Called `/usr/bin/pkg-config --modversion Magnum` -> 1

CMake binary for MachineChoice.HOST is not cached
None of 'CMAKE' are defined in the environment, not changing global flags.
CMake binary missing from cross or native file, or env var undefined.
Trying a default CMake fallback at cmake
Trying CMake binary cmake for machine MachineChoice.HOST at ['/usr/bin/cmake']
Found CMake: /usr/bin/cmake (3.18.1)
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.
Extracting basic cmake information
Try CMake generator: auto
None of 'CC' are defined in the environment, not changing global flags.
None of 'CFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
Called `/usr/bin/cmake --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt .` in /home/alp/practice/magnum/build/meson-private/cmake_Magnum -> 0
  -- Module search paths:    ['/', '/opt', '/usr', '/usr/local']
  -- CMake root:             /usr/share/cmake-3.18
  -- CMake architectures:    []
  -- CMake lib search paths: ['lib', 'lib32', 'lib64', 'libx32', 'share']

Determining dependency 'Magnum' with CMake executable '/usr/bin/cmake'
Try CMake generator: auto
None of 'CC' are defined in the environment, not changing global flags.
None of 'CFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
Called `/usr/bin/cmake -DNAME=Magnum -DARCHS= -DCOMPS= --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt .` in /home/alp/practice/magnum/build/meson-private/cmake_Magnum -> 1
CMake failed for generator  and package Magnum with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- Found Corrade: /usr/include  found components: Containers rc Utility 
-- Configuring incomplete, errors occurred!
See also "/home/alp/practice/magnum/build/meson-private/cmake_Magnum/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/share/cmake/Corrade/UseCorrade.cmake:43 (message):
  Corrade cannot be used with GCC < 4.8.1.  Sorry.
Call Stack (most recent call first):
  /usr/share/cmake/Corrade/FindCorrade.cmake:563 (include)
  /usr/share/cmake/Corrade/CorradeConfig.cmake:26 (include)
  /usr/share/cmake/Magnum/FindMagnum.cmake:223 (find_package)
  /usr/share/cmake/Magnum/MagnumConfig.cmake:26 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: auto
None of 'CC' are defined in the environment, not changing global flags.
None of 'CFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
CMake failed for generator  and package Magnum with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- Found Corrade: /usr/include  found components: Containers rc Utility 
-- Configuring incomplete, errors occurred!
See also "/home/alp/practice/magnum/build/meson-private/cmake_Magnum/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/share/cmake/Corrade/UseCorrade.cmake:43 (message):
  Corrade cannot be used with GCC < 4.8.1.  Sorry.
Call Stack (most recent call first):
  /usr/share/cmake/Corrade/FindCorrade.cmake:563 (include)
  /usr/share/cmake/Corrade/CorradeConfig.cmake:26 (include)
  /usr/share/cmake/Magnum/FindMagnum.cmake:223 (find_package)
  /usr/share/cmake/Magnum/MagnumConfig.cmake:26 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: Ninja
None of 'CC' are defined in the environment, not changing global flags.
None of 'CFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
Called `/usr/bin/cmake -G Ninja -DNAME=Magnum -DARCHS= -DCOMPS= --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt .` in /home/alp/practice/magnum/build/meson-private/cmake_Magnum -> 1
CMake failed for generator Ninja and package Magnum with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- Found Corrade: /usr/include  found components: Containers rc Utility 
-- Configuring incomplete, errors occurred!
See also "/home/alp/practice/magnum/build/meson-private/cmake_Magnum/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/share/cmake/Corrade/UseCorrade.cmake:43 (message):
  Corrade cannot be used with GCC < 4.8.1.  Sorry.
Call Stack (most recent call first):
  /usr/share/cmake/Corrade/FindCorrade.cmake:563 (include)
  /usr/share/cmake/Corrade/CorradeConfig.cmake:26 (include)
  /usr/share/cmake/Magnum/FindMagnum.cmake:223 (find_package)
  /usr/share/cmake/Magnum/MagnumConfig.cmake:26 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: Unix Makefiles
None of 'CC' are defined in the environment, not changing global flags.
None of 'CFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
Called `/usr/bin/cmake -G Unix Makefiles -DNAME=Magnum -DARCHS= -DCOMPS= --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt .` in /home/alp/practice/magnum/build/meson-private/cmake_Magnum -> 1
CMake failed for generator Unix Makefiles and package Magnum with error code 1
OUT:
Running with expanded trace output on.
Not searching for unused variables given on the command line.
Trace will be written to cmake_trace.txt
-- Found Corrade: /usr/include  found components: Containers rc Utility 
-- Configuring incomplete, errors occurred!
See also "/home/alp/practice/magnum/build/meson-private/cmake_Magnum/CMakeFiles/CMakeOutput.log".



ERR:
CMake Error at /usr/share/cmake/Corrade/UseCorrade.cmake:43 (message):
  Corrade cannot be used with GCC < 4.8.1.  Sorry.
Call Stack (most recent call first):
  /usr/share/cmake/Corrade/FindCorrade.cmake:563 (include)
  /usr/share/cmake/Corrade/CorradeConfig.cmake:26 (include)
  /usr/share/cmake/Magnum/FindMagnum.cmake:223 (find_package)
  /usr/share/cmake/Magnum/MagnumConfig.cmake:26 (include)
  CMakeLists.txt:15 (find_package)





Try CMake generator: Visual Studio 10 2010
None of 'CC' are defined in the environment, not changing global flags.
None of 'CFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
Called `/usr/bin/cmake -G Visual Studio 10 2010 -DNAME=Magnum -DARCHS= -DCOMPS= --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt .` in /home/alp/practice/magnum/build/meson-private/cmake_Magnum -> 1
CMake failed for generator Visual Studio 10 2010 and package Magnum with error code 1
OUT:



ERR:
CMake Error: Could not create named generator Visual Studio 10 2010

Generators
* Unix Makefiles               = Generates standard UNIX makefiles.
  Green Hills MULTI            = Generates Green Hills MULTI files
                                 (experimental, work-in-progress).
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  Watcom WMake                 = Generates Watcom WMake makefiles.
  CodeBlocks - Ninja           = Generates CodeBlocks project files.
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files.
  CodeLite - Ninja             = Generates CodeLite project files.
  CodeLite - Unix Makefiles    = Generates CodeLite project files.
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files.
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files.
  Kate - Ninja                 = Generates Kate project files.
  Kate - Unix Makefiles        = Generates Kate project files.
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files.




Run-time dependency magnum found: NO (tried pkgconfig and cmake)

meson.build:3:0: ERROR: Dependency "Magnum" not found, tried pkgconfig and cmake

**Comment by nirbheek** [comment]
I think https://github.com/mesonbuild/meson/pull/7549 will fix this, can you try with that?

**Comment by astinus87** [comment]
@nirbheek No #7549 doesn't fix it. I get the same error.

**Comment by mensinda** [comment]
This should now be fixed in meson 0.55.1. Can you please check this is the case?

**Comment by astinus87** [comment]
@mensinda Yes, it is fixed in 0.55.1.
